website/src/components/Edit/SequencesForm.tsx

```diff
@@ -1,117 +1,178 @@
 import { type Dispatch, type FC, type SetStateAction } from 'react';
 
-import { type KeyValuePair, type Row } from './InputField';
-import { mapErrorsAndWarnings, type SequenceEntryToEdit } from '../../types/backend.ts';
+import { type SequenceEntryToEdit } from '../../types/backend.ts';
+import type { ReferenceGenomesLightweightSchema } from '../../types/referencesGenomes.ts';
 import { FileUploadComponent } from '../Submission/FileUpload/FileUploadComponent.tsx';
 import { PLAIN_SEGMENT_KIND, VirtualFile } from '../Submission/FileUpload/fileProcessing.ts';
 
-function generateAndDownloadFastaFile(
-    accessionVersion: string,
-    sequenceData: string,
-    segmentKey?: string,
-    isSingleSegment: boolean = false,
-) {
-    const fileName =
-        isSingleSegment || !segmentKey ? `${accessionVersion}.fasta` : `${accessionVersion}_${segmentKey}.fasta`;
-
-    const header = isSingleSegment || !segmentKey ? accessionVersion : `${accessionVersion}_${segmentKey}`;
-
-    const fileContent = `>${header}\n${sequenceData}`;
+function generateAndDownloadFastaFile(fastaHeader: string | null, sequenceData: string | null) {
+    let fileContent = '';
+    let trimmedHeader = '';
+
+    if (fastaHeader === null && sequenceData === null) {
```

Comment:
Why does this case exist here? If there is nothing to download, then we shouldn't need a download function?